### ISSUE TITLE
Spiral lift hill top does not count as a block section

### ISSUE THREAD

**Issue by Gymnasiast** [OpenRCT2/OpenRCT2#12632]
Originally reported here: https://forums.openrct2.org/topic/5249-spiral-lift-block-issue/

To fix this, we would need to introduce two new track pieces to distinguish them from regular, non-top pieces. This will either need a new save format or conversion at runtime. It will also need the track piece ID to use a 16-bit integer throughout, as it might still use an 8-bit integer in places (or a fake 16-bit integer that uses bit 8 as a flag).

**Comment by duncanspumpkin** [comment]
Why would we need to add a new track piece. We already do this for normal chain lifts all we need to do is add additional code for spiral lifts.

**Comment by Gymnasiast** [comment]
@duncanspumpkin For normal chain lifts, we check if the piece is a `X_UP_TO_FLAT` piece with a chain. Meanwhile, curved lift hills do not have a different piece on top, it's the same as the intermediate section. How would you distinguish between the two?

**Comment by duncanspumpkin** [comment]
We check for the intermediate section. i.e. a none spiral lift hill piece. How were you imagining it would work with another track piece? That would be considerably harder to pull off rather than a simple check for a none spiral lift hill.

**Comment by Gymnasiast** [comment]
I don't think you really understand. Just build a spiral roller coaster and use the tile inspector on the pieces and the problem should be obvious immediately: a curved lift hill consists of **identical** pieces, unlike a regular one.

**Comment by ocalhoun6** [comment]
Is it possible to have the code look at the next track piece in line? In that case, any spiral lift hill followed by a piece that _isn't_ a spiral lift hill should count as a block brake.

**Comment by Gymnasiast** [comment]
The disadvantage of that is that it makes the code more complex and potentially also slower.

**Comment by ocalhoun6** [comment]
I'm no coding expert, but I'd do it something like:

```
if(current piece is a spiral lift hill){
     if(current ride is in block section mode){
          code to check next track piece and see if it's also a spiral lift hill 
               (could probably be combined into the next conditional so you don't have to set a new variable)
          if(next piece is not a spiral lift hill){
               designate current track piece as a block brake
          }
     }
}
```

That way, 99% all the code will be skipped except for that first conditional. Should be able to afford one conditional check, right? And the resource-intensive code would again be skipped if the ride's not in block section mode. So unless you build a park that's just _packed_ with block section spiral coasters, the resource-intensive code shouldn't be running too often and shouldn't be too much of a burden, right?

**Comment by duncanspumpkin** [comment]
I've looked at it and it's definitely more complex to add another track type. You would have to add something like the horrible station code to pull off what i think you want. The code is not nice to read or follow. Then you have all the extra position code, drawing code, placement code. In contrast we would just have to add one or two extra lines to the block unlock code and vehicle placement code. You are already iterating the track so it's not much more expensive to iterate it one extra section. Maybe I'm not seeing the issue you see though.
